src/Spout/Writer/Common/Entity/Worksheet.php

```diff
@@ -78,6 +95,79 @@ public function getMaxNumColumns()
         return $this->maxNumColumns;
     }
 
+    /**
+     * @return array
+     */
+    public function getColumnWidths()
+    {
+        return $this->columnWidths;
+    }
+
+    /**
+     * Gets the calculated max column width for the specified index
+     * @param int $zeroBasedIndex
+     * @return int
+     */
+    public function getMaxColumnWidth($zeroBasedIndex)
+    {
+        if (isset($this->columnWidths[$zeroBasedIndex])) {
+            return $this->columnWidths[$zeroBasedIndex];
+        }
+
+        $this->columnWidths[$zeroBasedIndex] = self::DEFAULT_COL_WIDTH;
+        return $this->columnWidths[$zeroBasedIndex];
+    }
+
+    /**
+     * Sets the calculated max column width for the specified index
+     * @param int $zeroBasedIndex
+     * @param int $value Value to set to
+     * @return void
+     */
+    public function setMaxColumnWidth($zeroBasedIndex, $value)
+    {
+        $curSize = $this->columnWidths[$zeroBasedIndex] ?? 0;
+        if ($curSize < $value) {
+            $this->columnWidths[$zeroBasedIndex] = $value;
+        }
+    }
+
+    /**
+     * Automatically calculates and sets the max column width for the specified cell
+     * @param Cell $cell The cell
+     * @param Style $style Row/Cell style
+     * @param int $zeroBasedIndex of cell
+     * @return void
+     */
+    public function autoSetWidth($cell, $style, $zeroBasedIndex)
+    {
+        $size = 1 + strlen($cell->getValue());//ensure we have at least 1 space
+        $size *= $style->isFontBold() ? 1.2 : 1.0;
```

Comment:
How did you come up with the 1.2 ratio?

Reply:
It's actually meant to be 700/400.. but the 1.2 seemed to fit better.. and that may be because tinier letters like i like you rightly pointed out occupy less space..